File: egghead.io/data-structures-and-algorithms-in-js/lesson-02-queue-data-structure-in-js/queue.js
class Queue {
  constructor() { 
    this.queue = []; 
  }

  enqueue(item) { 
    this.queue.unshift(item); 
  }

  dequeue() { 
    this.queue.pop(); 
  }

  peek() { 
    return this.queue[this.queue.length - 1]; 
  }

  get length() { 
    return this.queue.length;
  }

  isEmpty() { 
    return this.queue.length === 0; 
  }
}

module.exports = Queue;

File: egghead.io/data-structures-and-algorithms-in-js/lesson-02-queue-data-structure-in-js/queue.py
class Queue:
  def __init__(self):
    self.queue = [];

  def enqueue(self, item):
    self.queue.insert(0, item)

  def dequeue(self):
    self.queue.pop()

  def peek(self):
    return self.queue[len(self.queue) - 1]

  def length(self):
    return len(self.queue)

  def is_empty(self):
    return len(self.queue) == 0